File: MasterDirectory/MasterDirectory.Web/Modules/ServerTypes/Expansion/CategoriaExpansionForm.ts
import { StringEditor, LookupEditor, DateEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface CategoriaExpansionForm {
    LocalSap: StringEditor;
    LocationType: LookupEditor;
    Farmacia: StringEditor;
    FechaApertura: DateEditor;
    Reapertura: DateEditor;
    Comsuc: StringEditor;
    TipoEstaciona: LookupEditor;
    NCajonesEstaciona: StringEditor;
    Ciudad: StringEditor;
    Estado: StringEditor;
    Direccion: StringEditor;
    NExterior: StringEditor;
    Colonia: StringEditor;
    Cp: StringEditor;
    Latitud: StringEditor;
    Longuitud: StringEditor;
    FormatoFarmAlcance: LookupEditor;
    Pantallas: StringEditor;
    ProvMobiliario: LookupEditor;
    ColorMob: StringEditor;
    Dermo: StringEditor;
}

export class CategoriaExpansionForm extends PrefixedContext {
    static formKey = 'Expansion.CategoriaExpansion';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CategoriaExpansionForm.init)  {
            CategoriaExpansionForm.init = true;

            var w0 = StringEditor;
            var w1 = LookupEditor;
            var w2 = DateEditor;

            initFormType(CategoriaExpansionForm, [
                'LocalSap', w0,
                'LocationType', w1,
                'Farmacia', w0,
                'FechaApertura', w2,
                'Reapertura', w2,
                'Comsuc', w0,
                'TipoEstaciona', w1,
                'NCajonesEstaciona', w0,
                'Ciudad', w0,
                'Estado', w0,
                'Direccion', w0,
                'NExterior', w0,
                'Colonia', w0,
                'Cp', w0,
                'Latitud', w0,
                'Longuitud', w0,
                'FormatoFarmAlcance', w1,
                'Pantallas', w0,
                'ProvMobiliario', w1,
                'ColorMob', w0,
                'Dermo', w0
            ]);
        }
    }
}

File: MasterDirectory/MasterDirectory.Web/Modules/Expansion/CategoriaExpansion/CategoriaExpansionForm.cs
using Serenity.ComponentModel;
using System;

namespace MasterDirectory.Expansion.Forms;

[FormScript("Expansion.CategoriaExpansion")]
[BasedOnRow(typeof(CategoriaExpansionRow), CheckNames = true)]
public class CategoriaExpansionForm
{
    [Tab("General")]
    public string LocalSap { get; set; }
    //public string TipoFarmacia { get; set; }
    public string LocationType { get; set; }
    public string Farmacia { get; set; }
    public DateTime FechaApertura { get; set; }
    public DateTime? Reapertura { get; set; }
    public string Comsuc { get; set; }
    public string TipoEstaciona { get; set; }
    public string NCajonesEstaciona { get; set; }

    [Tab("Ubicacion")]
    public string Ciudad { get; set; }
    public string Estado { get; set; }
    public string Direccion { get; set; }
    public string NExterior { get; set; }
    public string Colonia { get; set; }
    public string Cp { get; set; }
    public string Latitud { get; set; }
    public string Longuitud { get; set; }

    [Tab("Mobiliario")]

    public string FormatoFarmAlcance { get; set; }
    public string Pantallas { get; set; }
    public string ProvMobiliario { get; set; }
    public string ColorMob { get; set; }
    public string Dermo { get; set; }
    //public DateTime DtRegistro { get; set; }
}